docs/CppInteroperability/SwiftToCppInteroperabilityVisionDocument.md

```diff
@@ -0,0 +1,137 @@
+
+## Overview of Swift's interoperability support
+
+Swift uses two very different approaches for interoperability with C and Objective-C. For "forward" interoperability, Swift embeds a copy of the Clang compiler, which it uses to directly load C and Objective-C Clang modules and translate declarations into a native Swift representation. For "reverse" interoperability, Swift does not want to require Clang to embed Swift, and so Swift generates an Objective-C header that can be used to call into Swift APIs that are exposed in that header. Because Objective-C is restricted in what kinds of types it can work with, and because it requires code to be emitted in a special way that doesn't match the regular Swift ABI, methods and types must be marked as being exposed to Objective-C with the `@objc` attribute.
+
+Swift's support for C++ interoperability is modeled after its support for C and Objective-C interoperability. For "forward" interoperability, the embedded Clang compiler is used to directly load C++ Clang modules and translate declarations into a native Swift representation. The "forward" interoperability [vision document](https://github.com/apple/swift/pull/60501/files) provides more details about this model. For "reverse" interoperability, Swift generates a header that uses C++ language constructs to represent Swift APIs that are exposed by the Swift module. Because C++ is much more expressive and flexible than Objective-C, the generated header is able to provide representation for native Swift functions, methods, initializers, accessors and types. This allows C++ programmers to call into Swift APIs using the familiar C++ function and member function call syntax.
+
+### C++ interoperability is opt-out not opt-in
+
+The user model employed by Swift-to-C++ interoperability is different than the user model employed by the Objective-C interoperability. C++ can use Swift's native calling convention when calling Swift functions, and C++ does not have restrictions on what Swift types it can work with. This means that Swift APIs do not require additional annotations when they're bridged to C++. This fact highlights the key difference between C++ and Objective-C interoperability; Swift library author is not required to change their code when their APIs are exposed to C++, whereas they are required to opt-in into interoperability using an `@objc` annotation when exposing APIs to Objective-C. Swift library authors will still be able to opt-out from exposing certain APIs to C++ when they desire to do so.
+
+The permissive user model of Swift-to-C++ interoperability implies that any Swift library that can be imported into Swift can also be "imported" into C++, as long as its APIs are representable by the C++ code in the generated header. The Swift compiler can generate this header from the library's module interface file upon user's request, even when the library's author did not have such use case in mind.
+
+## Goals
+
```

Comment:
The Language Workgroup feels that it would be appropriate to lead this section with the high-level goal, explaining why this is interesting to do at all.  Consider starting with something like the two paragraphs I suggested for the forward vision [here](https://github.com/apple/swift/pull/60501#discussion_r1036568365).

Reply:
I have added a lead paragraph. I also liked the second paragraph you suggested in the other PR about not changing Swift's goals, but I moved it over to the "Swift language evolution and API design should be unaffected" section.